### ISSUE TITLE
Feature Request: Make spectrogram params settable in Settings

### ISSUE THREAD

**Issue by drscotthawley** [artandlogic/vibrary#4]
In SettingsView, in addition to Username and Server, it'd be great it we also had...

Duration: length of audio clip in seconds. Defaults to 3.
(and maybe pull from the middle of the audio file instead of the beginning?) 

Channels: number of audio channels (1=mono, 2=stereo,...) Defaults to 1.

Mel Bins: number of mel-frequency bins.  Defaults to (whatever NumMels is already).

I figured out how to add the GUI elements for this, but haven't yet figured out where the defaults are supplied (e.g. 'hinton' for Server), or how to communicate these values to the code that implements them.

**Comment by jbagley** [comment]
We will always use a mono summation to input to the network won't we? 

SampleRate, I believe we handle sample rates automatically without needing the user to specify unless you mean you want to alter sample rates of training data. That would also be doable.

MelBins is not so easy because it requires changing the number of inputs to the network, and therefore needs to be stored with the sort file information as well as support for initializing the network properly. We don't build the graph, and when I last tried that it wasn't easy. 

Epochs shouldn't be hard. Vibrary just has to communicate it through the script it passes to panotti.

**Comment by drscotthawley** [comment]
We've always forced everything to mono for simplicity, yes, but Panotti can handle *arbitrary* numbers of channels -- like for those Norwegian navy cadets & their towed arrays, or my work with binaural audio classification. ;-)

For sample rate, we just want all the files to be resampled to the same rate, whatever that is.  But, for example you don't have to use 44.1 kHz, you could maybe choose to get by with just 16 kHz -- at the time of training, that is.  ...I bring this up because lots of (other) people in machine learning *do* use smaller sample rates than 44.1k, such as 16k or  22k.    


My thinking is that many of these parameters could be read from the Settings menu **only** for training, whereas for inference they could -- as you say -- be read from each sort file.  So then the sort file would in some sense "include instructions" for how to create the spectrograms to use for inference with it.

Parameters such as # of mel bins and # of channels are probably trivial to read directly from the dimensions of the input layer of the Tensorflow graph in the .pb file.  But Sample rate.... Panotti doesn't actually care about sample rate, so it knows nothing about it.  So either we pass sample rate to Panotti for when it outputs its file (?), or the Vibrary client somehow 'injects' this info into the .pb file after we download it from the server but before it gets used for inference.   ...?

**Comment by jbagley** [comment]
The audio and spectorgram characteristics would need to be stored in the sort file or someone changes this global setting and, potentially, accuracy goes down or fails altogether, doesn't it? The Training dialog could expose the options. Our sort files would become separate from the saved model, which is probably should be anyway, and include a path/URI to the saved model file.

**Comment by jbagley** [comment]
Potentially that information would live in the database rather than a file so users could some day query for stereo sort files vs mono sort files, 44.1K sort files or 16K sort files...

**Comment by jbagley** [comment]
Epochs could be a global setting.
